src/main/java/com/dataloom/datastore/services/LinkingService.java

```diff
@@ -80,7 +83,7 @@ public UUID link( UUID linkedEntitySetId, Set<Map<UUID, UUID>> linkingProperties
 
         // Warning: We assume that the restrictions on links are enforced/validated as specified in LinkingApi. In
         // particular, from now on we work on the assumption that only identical property types are linked on.
-        initializeComponents( linkIndexedByEntitySets.keySet(), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
+        initializeComponents( dsm.getLatestSyncId( linkIndexedByEntitySets.keySet() ), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
```

Comment:
(meant to leave this on the SearchService function part--I can also update the function though to take in a map instead if you need to generate the map anyway)